docs/tutorial/t-setup.md

```diff
@@ -0,0 +1,262 @@
+## Setup Environment for the Tutorial
+
+
+### Minimum requirements
+Before we start, make sure your machine meets the following requirements:
+* Ubuntu 22.04 (jammy) or later (the tutorial has been prepared and tested to work on 22.04)
+* 8 GB of RAM
+* 2 CPU threads
+* At least 20GB of available storage.
+* Access to the internet for downloading the required snaps and charms.
+
+### Prepare MicroK8s
+
+Charmed Spark is developed to be run on top of a Kubernetes cluster. For the purpose of this tutorial, we are going to use [MicroK8s](https://microk8s.io/), a very simple production-grade conformant K8s that can run locally. 
+
+Installing MicroK8s is as simple as running the following command:
+
+```bash
+sudo snap install microk8s --channel=1.28-strict/stable
+```
+
+Let's configure MicroK8s so that the currently logged in user has admin rights to the cluster.
+
+```bash
+# Set an alias 'kubectl' that can be used instead of microk8s.kubectl
+sudo snap alias microk8s.kubectl kubectl
+
+# Add the current user into 'microk8s' group
+sudo usermod -a -G snap_microk8s ${USER}
+
+# Create and provide ownership of '~/.kube' directory to current user
+mkdir -p ~/.kube
+sudo chown -f -R ${USER} ~/.kube
+
+# Put the group membership changes into effect
+newgrp snap_microk8s
+```
+
+Once done, the status of the MicroK8s can be verified with
+
+```bash
+microk8s status --wait-ready
+```
+
+When MicroK8s cluster is running and ready, you should see an output similar to the following:
+
+```
+microk8s is running
+high-availability: no
+...
+addons:
+  enabled:
+    dns                  # (core) CoreDNS
+    ha-cluster           # (core) Configure high availability on the current node
+    helm                 # (core) Helm - the package manager for Kubernetes
+    helm3                # (core) Helm 3 - the package manager for Kubernetes
+  disabled:
+    cert-manager         # (core) Cloud native certificate management
+...
+```
+
+Let's generate the Kubernetes configuration file using MicroK8s and write it to `~/.kube/config`. This is where Kubernetes looks for the Kubeconfig file by default.
+
+```bash
+microk8s config | tee ~/.kube/config
+```
+
+Now let's enable a few addons for using features like role based access control, load balancing and usage of local volume for storage.
+
+```bash
+sudo microk8s enable rbac storage hostpath-storage
+
+sudo apt install -y jq
+
+# Get IP address
+IPADDR=$(ip -4 -j route get 2.2.2.2 | jq -r '.[] | .prefsrc')
+sudo microk8s enable metallb:$IPADDR-$IPADDR
+```
```

Comment:
```suggestion
Now let's enable a few addons for using features like role based access control, usage of local volume for storage, and load balancing

```bash
# Enable rbac
sudo microk8s enable rbac

# Enable storage
sudo microk8s enable storage hostpath-storage

# Enabling metallb for load balancing
sudo apt install -y jq
IPADDR=$(ip -4 -j route get 2.2.2.2 | jq -r '.[] | .prefsrc')
sudo microk8s enable metallb:$IPADDR-$IPADDR
```
```